--- a/EPW/src/io/io.f90
+++ b/EPW/src/io/io.f90
@@ -548,13 +548,13 @@
     IF (etf_mem == 0) THEN
       !
       filint = TRIM(tmp_dir) // TRIM(prefix) // '.epmatwp'
       IF (TRIM(lsda) == 'down') filint = TRIM(tmp_dir) // TRIM(prefix) // '.down.epmatwp'
       !
+      ALLOCATE(epmatwp(nbndsub, nbndsub, nrr_k, nmodes, nirg_loc), STAT = ierr)
+      IF (ierr /= 0) CALL errore('epw_read', 'Error allocating epmatwp', 1)
+      epmatwp = czero
+      !
 #if defined(__MPI)
       !
-      ALLOCATE(epmatwp(nbndsub, nbndsub, nrr_k, nmodes, nirg_loc), STAT = ierr)
-      IF (ierr /= 0) CALL errore('epw_read', 'Error allocating epmatwp', 1)
-      epmatwp = czero
-      !
       CALL MPI_TYPE_CONTIGUOUS((nbndsub**2) * nrr_k, MPI_DOUBLE_COMPLEX, epmatwp_block_dtype, ierr)
       IF (ierr /= 0) CALL errore('epw_read', 'Error creating epmatwp_block_dtype', 1)
